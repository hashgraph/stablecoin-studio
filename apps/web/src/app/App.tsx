import { I18nextProvider, useTranslation } from 'react-i18next';
import { ChakraProvider, Flex, Text } from '@chakra-ui/react';
import { Provider } from 'react-redux';
import i18n from '../i18n';
import store from '../store/store';
import theme from '../theme/Theme';
import Router from '../Router/Router';
import { BrowserRouter } from 'react-router-dom';
import { Fonts } from '../components/Fonts';
import { Focus } from '../components/Focus';
import { ScrollBar } from '../components/Scrollbar';
import InnactivityTimer from '../components/InnactivityTimer';
import { isMobile } from 'react-device-detect';
import { useState } from 'react';
import Disclaimer from './Disclaimer';

function App() {
	const { t } = useTranslation('global');
	const [accepted, setAccepted] = useState<boolean>(false);
	const showDisclaimer: boolean =
		process.env.REACT_APP_SHOW_DISCLAIMER !== undefined &&
		process.env.REACT_APP_SHOW_DISCLAIMER === 'true';

	return isMobile ? (
		<Flex h='100vh' justify={'center'} flexDir='column'>
			<Text
				fontSize='16px'
				fontWeight={500}
				textAlign='center'
				lineHeight='16px'
				color='brand.gray'
				data-testid='isMobile'
			>
				{t('mobile.message')}
			</Text>
		</Flex>
	) : !showDisclaimer || accepted ? (
		<I18nextProvider i18n={i18n}>
			<Provider store={store}>
				<ChakraProvider theme={theme}>
					<BrowserRouter>
						<InnactivityTimer>
							<Focus />
							<Fonts />
							<ScrollBar />
							<Router />
						</InnactivityTimer>
					</BrowserRouter>
				</ChakraProvider>
			</Provider>
		</I18nextProvider>
	) : (
		<Disclaimer setAccepted={setAccepted} />
	);
}

export default App;
