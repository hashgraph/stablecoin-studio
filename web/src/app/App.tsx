import { I18nextProvider } from 'react-i18next';
import { ChakraProvider } from '@chakra-ui/react';
import { Provider } from 'react-redux';
import i18n from '../i18n';
import store from '../store/store';
import theme from '../theme/Theme';
import Router from '../Router/Router';
import { BrowserRouter } from 'react-router-dom';
import { Fonts } from '../components/Fonts';
import { Focus } from '../components/Focus';

function App() {
	return (
		<I18nextProvider i18n={i18n}>
			<Provider store={store}>
				<ChakraProvider theme={theme}>
					<BrowserRouter>
						<Focus />
						<Fonts />
						<Router />
					</BrowserRouter>
				</ChakraProvider>
			</Provider>
		</I18nextProvider>
	);
}

export default App;
