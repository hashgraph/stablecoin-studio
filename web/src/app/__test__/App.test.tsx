import { render } from '@testing-library/react';
import App from '../App';

describe(`<${App.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<App />, {});

		expect(component.asFragment()).toMatchSnapshot();
	});
});
