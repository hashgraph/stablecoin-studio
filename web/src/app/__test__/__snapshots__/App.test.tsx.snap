// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<DocumentFragment>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
